File: src/main/java/io/stouder/adonis/cli/json/ace/CommandFlag.java
package io.stouder.adonis.cli.json.ace;

import com.google.gson.annotations.SerializedName;
import lombok.Getter;

@Getter
public class CommandFlag<T> {
    private String name;
    private String flagName;
    private boolean required;
    private String type;
    private String description;
    @SerializedName("default")
    private T defaultValue;
}

File: src/main/java/io/stouder/adonis/cli/json/ace/CommandFlag.kt
package io.stouder.adonis.cli.json.ace

import com.google.gson.annotations.SerializedName

data class CommandFlag<T>(
    val name: String,
    val flagName: String,
    val required: Boolean,
    val type: String,
    val description: String,
    @SerializedName("default") val defaultValue: T
)